### ISSUE TITLE
Fix Molly's ability to backup over Nextcloud/WebDAV

### ISSUE THREAD

**Issue by haarp** [mollyim/mollyim-android#396]
### Is there an existing request for this?

- [X] I have searched the existing issues

### Feature description

Hello,

when Signal is writing its backup, it first creates a temporary file and then renames it. This rename dance appears to throw off some tools, namely Nextcloud and DAVx5's WebDAV share function. The result is Signal aborting the backup and partially uploaded files. I tested both apps on a recent Android 15 device, both failed to allow Signal/Molly to finish its backup . Now it's possible that both of these tools are simply buggy, but Signal is definitely not innocent either.

This is a [known issue](https://github.com/signalapp/Signal-Android/issues/10258) that has been around for years. Despite numerous affected people, Signal does not seem interestd in fixing it.

The fix could be as simple as:

- Not create a temporary file, write into the final filename directly
- or create the temporary file in a different location and then copy it over
- or tweaking the timings to avoid whatever confuses these sync tools

Would Molly be interested in tackling this? Thanks a lot!

**Comment by randomwithnoname** [comment]
Hello! Signal is overhauling their backups system and it seems they will be selling cloud storage. Not to worry, Molly will continue to allow local backups and make it practical for use with any cloud provider: https://github.com/mollyim/mollyim-android-unifiedpush/pull/20.

**Comment by HaleTom** [comment]
Related: #224 #544

**Comment by p1gp1g** [comment]
This is finally possible with v8.7.3-1:

Settings > Backups > On-device backups offers the new incremental backup, and you can use a [SAF](https://developer.android.com/guide/topics/providers/document-provider) folder (eg. a WebDAV mount point using DAVx5 or another cloud application)

**Comment by haarp** [comment]
> This is finally possible with v8.7.3-1:
> 
> Settings > Backups > On-device backups offers the new incremental backup, and you can use a [SAF](https://developer.android.com/guide/topics/providers/document-provider) folder (eg. a WebDAV mount point using DAVx5 or another cloud application)

Indeed, it's doing incremental backups now. There's no toggle for anyone wondering.

But such a backup somehow triggered the Nextcloud app to go haywire, continuously trying to upload non-existent files and needing a data reset to settle down. But that's just Nextcloud being Nextcloud. It's app-shaped delirium.

Backups work really well over DAVx5 (webdav)!

Thanks for your implementation of this!
